### PR TITLE
Validation script

### PR DESCRIPTION
**Pull request by techlake**

### CODE REVIEW

File: Submit-PolicyDefinitionFile.ps1 [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        $idSplits = $definition.id -split "/"
+        $name = $idSplits[-1]
+    }
+    elseif ($properties.id) {
+        $idSplits = $properties.id -split "/"
+        $name = $idSplits[-1]
+    }
+}
+if ($name) {
+    if (!([guid]::TryParse($name, $([ref][guid]::Empty)))) {
```
Comment (DonKoning) [inline Submit-PolicyDefinitionFile.ps1:43]: $name = "" and therefore new GUID is created every time.
type [out] is unknown, if I replace with $([ref][guid]::Empty), it works for me.

File: Submit-PolicyDefinitionFile.ps1 [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+if (!$displayName) {
+    Write-Error "displayName not found."
+}
+$description = $properties.description
+if (!$description) {
+    Write-Error "description not found."
+}
+
+$version = $properties.version
+# if (!$version) {
```
Comment (DonKoning) [inline Submit-PolicyDefinitionFile.ps1:64]: Why leave commented-out code in?

Not following logic here:
$version = $properties.version (doesn't exist)

If version is missing, it's not added to the metadata property.

File: Submit-PolicyDefinitionFile.ps1 [hunk 3]
```diff
@@ -0,0 +1,306 @@
+#Requires -PSEdition Core
```
Comment (DonKoning) [inline Submit-PolicyDefinitionFile.ps1:1]: Please add comment-based help

**Review by DonKoning — CHANGES_REQUESTED** [review]

File: Submit-PolicyDefinitionFile.ps1 (hunk 1 above)
Comment (techlake) [reply Submit-PolicyDefinitionFile.ps1:43]: This was fixed. I will kill the PR and reissue

File: Submit-PolicyDefinitionFile.ps1 (hunk 2 above)
Comment (techlake) [reply Submit-PolicyDefinitionFile.ps1:64]: This is needed due to the current state of version support by PG. This is about the new version, not the version in metadata.
Would you want a check on the metadata for now?

File: Submit-PolicyDefinitionFile.ps1 (hunk 3 above)
Comment (techlake) [reply Submit-PolicyDefinitionFile.ps1:1]: OK

**Comment by techlake** [comment]
Need to reissue, something strange happened with my commits

File: Submit-PolicyDefinitionFile.ps1 (hunk 2 above)
Comment (DonKoning) [reply Submit-PolicyDefinitionFile.ps1:64]: Yes, let's

**Comment by techlake** [comment]
Done

Heinrich Gantenbein

**Pull request closed without merging**